chrome/content/zotero/elements/itemPane.js

```diff
@@ -0,0 +1,522 @@
+			return this._sidenav;
+		}
+
+		set sidenav(sidenav) {
+			this._sidenav = sidenav;
+		}
+
+		static get observedAttributes() {
+			return ['pinnedPane'];
+		}
+
+		init() {
+			this._container = this.querySelector('#zotero-view-item-container');
+			this._header = this.querySelector('#zotero-item-pane-header');
+			this._paneParent = this.querySelector('#zotero-view-item');
+
+			this._container.addEventListener("keypress", this._handleKeypress);
+			this._paneParent.addEventListener('scroll', this._handleContainerScroll);
+
+			this._paneHiddenOb = new MutationObserver(this._handlePaneStatus);
+			this._paneHiddenOb.observe(this._paneParent, {
+				attributes: true,
+				attributeFilter: ["hidden"],
+				subtree: true,
+			});
+			this._initIntersectionObserver();
+
+			this._unregisterID = Zotero.Notifier.registerObserver(this, ['item'], 'itemPane');
+
+			this._disableScrollHandler = false;
+			this._pinnedPaneMinScrollHeight = 0;
+		}
+
+		destroy() {
+			this._container.removeEventListener("keypress", this._handleKeypress);
+			this._paneParent.removeEventListener('scroll', this._handleContainerScroll);
+
+			this._paneHiddenOb.disconnect();
+			this._intersectionOb.disconnect();
+
+			Zotero.Notifier.unregisterObserver(this._unregisterID);
+		}
+
+		async render() {
+			if (!this.initialized || !this.item) {
+				return;
+			}
+			let item = this.item;
+			Zotero.debug('Viewing item');
+			this._isRendering = true;
+
+			let panes = this.getPanes();
+			let pendingBoxes = [];
+			let inTrash = ZoteroPane.collectionsView.selectedTreeRow && ZoteroPane.collectionsView.selectedTreeRow.isTrash();
+			for (let box of [this._header, ...panes]) {
+				if (!box.showInFeeds && item.isFeedItem) {
+					box.style.display = 'none';
+					box.hidden = true;
+					continue;
+				}
+				else {
+					box.style.display = '';
+					box.hidden = false;
+				}
+				
+				if (this.mode) {
+					box.mode = this.mode;
+					
+					if (box.mode == 'view') {
+						box.hideEmptyFields = true;
+					}
+				}
+				else {
+					box.mode = 'edit';
+				}
+				
+				box.item = item;
+				box.inTrash = inTrash;
+				// Render sync boxes immediately
+				if (!box.hidden && box.render) {
+					if (box.render instanceof AsyncFunction) {
+						pendingBoxes.push(box);
+					}
+					else {
+						box.render();
+					}
+				}
+			}
+
+			let pinnedPaneElem = this.getPane(this.pinnedPane);
+			let pinnedIndex = panes.indexOf(pinnedPaneElem);
+			
+			this._paneParent.style.paddingBottom = '';
+			await waitFrames(3);
+			if (pinnedPaneElem) {
+				let paneID = pinnedPaneElem.dataset.pane;
+				this.scrollToPane(paneID, 'instant');
+				this.pinnedPane = paneID;
+			}
+			else {
+				this._paneParent.scrollTo(0, 0);
+			}
+
+			// Only render visible panes
+			for (let box of pendingBoxes) {
+				if (pinnedIndex > -1 && panes.indexOf(box) < pinnedIndex) {
+					continue;
+				}
+				if (!this.isPaneVisible(box.dataset.pane)) {
+					continue;
+				}
+				await box.render();
```

Comment:
After scrolling, check all boxes that have pending rendering tasks. Only those that are visible will be called. The invisible boxes are rendered when scrolling into view.